appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -1,32 +1,135 @@
 package com.microsoft.appcenter.reactnative.data;
 
-import android.widget.Toast;
+import android.app.Application;
 
-import com.facebook.react.bridge.NativeModule;
-import com.facebook.react.bridge.ReactApplicationContext;
-import com.facebook.react.bridge.ReactContext;
-import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.BaseJavaModule;
+import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
 
-import java.util.Map;
-import java.util.HashMap;
+import com.facebook.react.bridge.WritableNativeArray;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
 
-public class AppCenterReactNativeDataModule extends ReactContextBaseJavaModule {
+import com.microsoft.appcenter.AppCenter;
+import com.microsoft.appcenter.data.Data;
+import com.microsoft.appcenter.data.models.DocumentWrapper;
+import com.microsoft.appcenter.reactnative.shared.AppCenterReactNativeShared;
+import com.microsoft.appcenter.utils.async.AppCenterConsumer;
 
-  private static final String DURATION_SHORT_KEY = "SHORT";
-  private static final String DURATION_LONG_KEY = "LONG";
 
-  public AppCenterReactNativeDataModule(ReactApplicationContext reactContext) {
-    super(reactContext);
-  }
+public class AppCenterReactNativeDataModule extends BaseJavaModule {
 
-  @Override
-  public String getName() {
-    return "AppCenterReactNativeData";
-  }
+    public AppCenterReactNativeDataModule(Application application, boolean startEnabled) {
+        AppCenterReactNativeShared.configureAppCenter(application);
 
-  @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
-  }
+        if (AppCenter.isConfigured()) {
+            AppCenter.start(Data.class);
+
+            if (!startEnabled) {
```

Comment:
What is the use case of starting disabled? Was that discussed?

Reply:
Removed